### PR TITLE
save last model after saving top_k when save_last=True

### LINKED ISSUE #2680: Checkpoint saving order
The last model save action, [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/1369012bc71f257dcf7423ec65146d055ddc1cc7/pytorch_lightning/callbacks/model_checkpoint.py#L295), should be after saving the top k model because the best model and best score could have changed. Swapping the order allows resuming the training from the last checkpoint, with the last checkpoint having the latest information about the best model path/score.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #2680

The checkpoint named "last.ckpt" should be saved last, literally, after saving all the top_k, because these will update the state for `best_model_score` and `best_model_path`, and if we don't save last these infos won't land in the checkpoint.

Note: **Test fails on master**

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 from pytorch_lightning.callbacks.base import Callback
 from pytorch_lightning.utilities import rank_zero_warn, rank_zero_only
 
+CHECKPOINT_NAME_LAST = "last.ckpt"
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:20]: shall this be in class?

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -340,6 +340,10 @@ def on_validation_end(self, trainer, pl_module):
             assert trainer.global_rank == 0, 'tried to make a checkpoint from non global_rank=0'
             self._save_model(filepath, trainer, pl_module)
 
+        if self.save_last:
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:343]: it seems that the code was just moved or am I missing something?

File: tests/callbacks/test_model_checkpoint.py [hunk 3]
```diff
@@ -93,3 +99,24 @@ def test_model_checkpoint_no_extraneous_invocations(tmpdir):
...
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        early_stop_callback=False,
+        checkpoint_callback=model_checkpoint,
+        max_epochs=num_epochs,
+    )
+    trainer.fit(model)
+    path_last_epoch = model_checkpoint.format_checkpoint_name(num_epochs - 1, {})
+    path_last = tmpdir / CHECKPOINT_NAME_LAST
+    ckpt_last_epoch = torch.load(str(path_last_epoch))
```
Comment (Borda) [inline tests/callbacks/test_model_checkpoint.py:119]: let's add assert that the checkpoint is not the same
aka `path_last_epoch != path_last`

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/callbacks/model_checkpoint.py:343]: yes, that's the fix! it makes a difference.

File: tests/callbacks/test_model_checkpoint.py (hunk 3 above)
Comment (awaelchli) [reply tests/callbacks/test_model_checkpoint.py:119]: good idea!!!

Comment (awaelchli) [reply tests/callbacks/test_model_checkpoint.py:120]: done :)

File: pytorch_lightning/callbacks/model_checkpoint.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/callbacks/model_checkpoint.py:20]: ok!

**Pull request merged**